node-zerox/src/utils.ts

```diff
@@ -234,6 +261,28 @@ export const convertPdfToImages = async ({
     const convertResults = await storeAsImage.bulk(pagesToConvertAsImages, {
       responseType: "buffer",
     });
+    const numRequiredWorkers = convertResults.length * 4;
+    let numNewWorkers = numRequiredWorkers - NUM_STARTING_WORKERS;
+
+    if (maxTesseractWorkers !== -1) {
+      const numPreviouslyInitiatedWorkers =
+        maxTesseractWorkers < NUM_STARTING_WORKERS
+          ? maxTesseractWorkers
+          : NUM_STARTING_WORKERS;
+
+      if (numRequiredWorkers > numPreviouslyInitiatedWorkers) {
+        numNewWorkers = Math.min(
+          numRequiredWorkers - numPreviouslyInitiatedWorkers,
+          maxTesseractWorkers - numPreviouslyInitiatedWorkers
+        );
+      } else {
+        numNewWorkers = 0;
+      }
+    }
+
+    // Add more workers if needed
+    if (numNewWorkers > 0) initTesseractScheduler(numNewWorkers);
```

Comment:
Do we need to `await` here?

Reply:
Nope, the goal is to have this run in parallel. The scheduler assigns a job to the worker as soon as its added (if a job is available).